src/voice_registration.js

```diff
@@ -55,154 +66,77 @@ go.app = function() {
-            });
-        });
-
-        self.states.add('state_r08_pregnant_nextyear_month', function(name) {
-            // TODO #6 Dynamically generate months
-            var routing = {
-                'january': 'state_r11_pregnant_day',
-                'february': 'state_r11_pregnant_day',
-                'march': 'state_r11_pregnant_day',
-                'april': 'state_r11_pregnant_day',
-                'may': 'state_r11_pregnant_day'
-            };
+                next: 'state_r05_birth_year'
+            });
+        });
+
+        self.states.add('state_r05_birth_year', function(name) {
+            // TODO #6 Don't show next year for pregnancy in Jan / Feb
+            var speech_option;
+            var year_choices = [
+                new Choice('last_year', $('last_year')),
+                new Choice('this_year', $('this_year')),
+                new Choice('next_year', $('next_year'))
+            ];
+            if (self.im.user.answers.state_r04_mom_state === 'pregnant') {
+                choices = year_choices.slice(1,3);
+                speech_option = '1';
+            } else {
+                choices = year_choices.slice(0,2);
+                speech_option = '2';
+            }
             return new ChoiceState(name, {
-                question: $('Which month?'),
-                choices: [
-                    new Choice('january', $('january')),
-                    new Choice('february', $('february')),
-                    new Choice('march', $('march')),
-                    new Choice('april', $('april')),
-                    new Choice('may', $('may'))
-                ],
+                question: $('Birth year?'),
+                helper_metadata: go.utils.make_voice_helper_data(
+                    self.im, name, lang, speech_option),
+                choices: choices,
                 next: function(choice) {
-                    return routing[choice.value];
+                    return 'state_r06_birth_month';
                 }
             });
         });
 
-        self.states.add('state_r09_baby_lastyear_month', function(name) {
-            // TODO #6 Dynamically generate months
-            var routing = {
-                'july': 'state_r12_baby_day',
-                'august': 'state_r12_baby_day',
-                'september': 'state_r12_baby_day',
-                'october': 'state_r12_baby_day',
-                'november': 'state_r12_baby_day',
-                'december': 'state_r12_baby_day'
-            };
+        self.states.add('state_r06_birth_month', function(name) {
+            var speech_option;
+            self.im.user.answers.state_r04_mom_state === 'pregnant'
+                ? speech_option = '1'
+                : speech_option = '2';
+            // create choices eg. new Choice('1', $('1')) etc.
+            var month_choices = [];
+            for (i=1; i<=12; i++) {
+                month_choices.push(new Choice(i.toString(), $(i.toString())));
```

Comment:
I *think* this will break the translation engine. However, you don't need to translation the digits 1-12 :grinning: